### PR TITLE
docs: point users to docs.cilium.io in GH docs

### PR DESCRIPTION
**Pull request by aanm**
Since users might read the github docs we will provide them a warning
in the documentation, that will not show up in the generated html code
but it shows up in github, to point them into the stable documentation
in http://docs.cilium.io

Signed-off-by: André Martins <andre@cilium.io>

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/4627)

### CODE REVIEW

**Comment by aanm** [comment]
Unfortunately I can't do the same for the `.md` files in `Documentation/cmdref`

**Comment by aanm** [comment]
test-docs-please

**Comment by aanm** [comment]
should we backport it?

**Review by raybejjani — APPROVED** [review]
Do we expect deeplinks to the GH pages? most would land on the main README, wouldn't they?

I'm not opposed to this but I'm surprised anyone was expecting a GH page to be stable... shrug

**Comment by aanm** [comment]
test-docs-please

**Comment by tgraf** [comment]
@aanm I'm OK with this but needs an addition to undo this when we branch off a release branch from master

**Comment by aanm** [comment]
@tgraf this was intend to be a message that shows up in github but it does not show up in readthedocs regardless of which version it is. Which message will be equally informative and could stay on all versions that would push the users using readthedocs instead of github?

**Comment by tgraf** [comment]
> Which message will be equally informative and could stay on all versions that would push the users using readthedocs instead of github?

OK. The "latest" was misleading to me. I would reword it to:

```
+    WARNING: You are looking at unreleased Cilium documentation.
+    Please use the official rendered version released here:
       http://docs.cilium.io
```

**Comment by aanm** [comment]
test-me-please

**Review by ianvernon — APPROVED** [review]

**Review by tgraf — APPROVED** [review]
